src/gf.c

```diff
@@ -1810,18 +1826,21 @@ JL_DLLEXPORT void jl_method_table_insert(jl_methtable_t *mt, jl_method_t *method
             size_t i, l = jl_svec_len(specializations);
             for (i = 0; i < l; i++) {
                 jl_method_instance_t *mi = (jl_method_instance_t*)jl_svecref(specializations, i);
-                if (mi != NULL && !jl_has_empty_intersection(type, (jl_value_t*)mi->specTypes))
-                    if (invalidate_backedges(mi, max_world)) {
-                        invalidated = 1;
-                        if (_jl_debug_method_invalidation) {
-                            jl_array_ptr_1d_push(_jl_debug_method_invalidation, (jl_value_t*)mi);
-                            if (!loctag) {
-                                loctag = jl_cstr_to_string("jl_method_table_insert");
-                                jl_gc_wb(_jl_debug_method_invalidation, loctag);
+                if (mi != NULL) {
+                    isect = jl_type_intersection(type, (jl_value_t*)mi->specTypes);
+                    if (isect != jl_bottom_type && !is_call_ambiguous(mt, isect, max_world))
+                        if (invalidate_backedges(mi, max_world)) {
+                            invalidated = 1;
+                            if (_jl_debug_method_invalidation) {
+                                jl_array_ptr_1d_push(_jl_debug_method_invalidation, (jl_value_t*)mi);
+                                if (!loctag) {
+                                    loctag = jl_cstr_to_string("jl_method_table_insert");
+                                    jl_gc_wb(_jl_debug_method_invalidation, loctag);
```

Comment:
This `jl_gc_wb` doesn't do anything (there's no memory being modified). Why is it here?

Reply:
My fault, came from resolving conflict with #35768 and I resolved it into here.

Reply:
None of those do anything either. What are they supposed to do?

Reply:
What are "those"? You mean all the `jl_gc_wb` in #35768? I thought from https://docs.julialang.org/en/latest/manual/embedding/#Updating-fields-of-GC-managed-objects-1 that I'd need to protect them but I guess that's not necessary for array elements, only fields? I can get rid of them.

Reply:
Wouldn't work for that anyways